modules/core/src/webgl/classes/texture-3d.js

```diff
@@ -1,87 +1,18 @@
 import GL from '@luma.gl/constants';
 import Texture from './texture';
-import Buffer from './buffer';
-import {withParameters} from '../context';
 import {isWebGL2, assertWebGL2Context} from '../utils';
 
 export default class Texture3D extends Texture {
   static isSupported(gl) {
     return isWebGL2(gl);
   }
 
-  constructor(gl, opts = {}) {
+  constructor(gl, props = {}) {
     assertWebGL2Context(gl);
-    super(gl, Object.assign({}, opts, {target: opts.target || GL.TEXTURE_3D}));
+    props = Object.assign({}, props, {target: GL.TEXTURE_3D, unpackFlipY: false});
+    super(gl, props);
+    this.initialize(props);
 
-    this.width = null;
-    this.height = null;
-    this.depth = null;
     Object.seal(this);
-
-    this.setImageData(opts);
-    if (opts.generateMipmap) {
-      this.generateMipmap();
-    }
-  }
-
-  initialize(opts = {}) {
-    this.opts = Object.assign({}, this.opts, opts);
-    const {pixels, settings} = this.opts;
-    if (settings) {
-      withParameters(settings, () => {
-        if (pixels) {
-          this.setImage3D(this.opts);
-        }
-      });
-      this.setParameters(opts);
-    }
-  }
-
-  // WebGL2
-
-  // Image 3D copies from Typed Array or WebGLBuffer
-  setImage3D({
```

Comment:
Nit: `setImage3D()`: I supposed is also used by e.g. Texture2DArray? Though if this is only used by 3d textures then it could be better to keep it here as it will otherwise be bundled for all apps, even if they never use a 3D texture.